galasa-ui/src/utils/functions.ts

```diff
@@ -53,24 +62,36 @@ export function parseIsoDateTime(isoString: string) {
  * @returns formatted duration string
  */
 export function getIsoTimeDifference(startTime: string, endTime: string): string {
-  // Parse dates and get difference in seconds
-  const t1 = Date.parse(startTime);
-  const t2 = Date.parse(endTime);
-  let delta = Math.abs(t2 - t1) / 1000;  // seconds
-
-  // Compute components
-  const hours   = Math.floor(delta / 3600);
-  delta -= hours * 3600;
-  const minutes = Math.floor(delta / 60);
-  const seconds = Math.round((delta - minutes * 60) * 10) / 10;
-
-  // Build parts with proper pluralization, skipping zero units
-  const parts: string[] = [];
-  if (hours   > 0) parts.push(`${hours} hr${hours   !== 1 ? "s" : ""}`);
-  if (minutes > 0) parts.push(`${minutes} min${minutes !== 1 ? "s" : ""}`);
-  if (seconds > 0 || parts.length === 0) {
-    parts.push(`${seconds} sec${seconds !== 1 ? "s" : ""}`);
+
+  //Parse both timestamps to get time in milliseconds
+  const startedAt = Date.parse(startTime);
+  const endedAt   = Date.parse(endTime);
+
+  let result: string;
+
+  // If either parse failed, produce an error message
+  if (isNaN(startedAt) || isNaN(endedAt)) {
+    result = "Invalid date";
+  } else {
+    // Compute absolute delta in seconds
+    let delta = Math.abs(endedAt - startedAt) / 1000;
+
+    // Break into components
+    const hours   = Math.floor(delta / 3600);
+    delta         -= hours * 3600;
+    const minutes = Math.floor(delta / 60);
+    const seconds = Math.round((delta - minutes * 60) * 10) / 10;
+
+    // Skipping zero units (except we always show seconds if everything else is zero)
+    const parts: string[] = [];
+    if (hours   > 0) parts.push(`${hours} hr${hours   !== 1 ? "s" : ""}`);
+    if (minutes > 0) parts.push(`${minutes} min${minutes !== 1 ? "s" : ""}`);
+    if (seconds > 0 || parts.length === 0) {
+      parts.push(`${seconds} sec${seconds !== 1 ? "s" : ""}`);
+    }
```

Comment:
If this is formatting the datetimes, it could be a separate function?

Also there's some inconsistency in the formatting of `if` conditions here which makes it a bit confusing to read. Would be nice to have `{}` brackets around the `if` statements for consistency.